packages/core-data/src/fetch/__experimental-fetch-link-suggestions.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Filters the search by type
+ *
+ * @typedef { 'post' | 'term' | 'post-format' } WPLinkSearchType
+ */
+
+/**
+ * @typedef WPLinkSearchOptions
+ *
+ * @property {boolean}             [isInitialSuggestions] Displays initial search suggestions, when true.
+ * @property {WPLinkSearchType}    [type]                 Filters by search type.
+ * @property {string}              [subtype]              Slug of the post-type or taxonomy.
+ * @property {number}              [page]                 Which page of results to return.
+ * @property {number}              [perPageArg]           Search results per page.
+ */
+
+/**
+ * @typedef WPLinkSearchResult
+ *
+ * @property {number} id     Post or term id.
+ * @property {string} url    Link url.
+ * @property {string} title  Title of the link.
+ * @property {string} type   The taxonomy or post type slug or type URL.
+ */
+
+/**
+ * Fetches link suggestions from the API.
+ *
+ * @async
+ * @typedef {( searchText: string, searchOptions: WPLinkSearchOptions ) => Promise< WPLinkSearchResult[] >} WPFetchLinkSuggestions
+ */
+
+/**
+ * Returns a function that when invoked, fetches link suggestions from the API.
+ *
+ * @param {Object}   [editorSettings]
+ * @param {boolean}  [editorSettings.disablePostFormats]
+ *
+ * @example
+ * ```js
+ * import { __experimentalFetchLinkSuggestions as createFetchLinkSuggestions } from '@wordpress/core-data';
+ *
+ * //...
+ *
+ * export function initialize( id, settings ) {
+ *
+ * settings.__experimentalFetchLinkSuggestions = createFetchLinkSuggestions(
+		settings
+	);
+ * ```
+ * @return { WPFetchLinkSuggestions } Function that fetches link suggestions
+ *
+ */
+const createFetchLinkSuggestions = ( {
+	disablePostFormats = false,
+} = {} ) => async ( search, searchOptions ) => {
+	const {
+		isInitialSuggestions = false,
+		type = undefined,
+		subtype = undefined,
+		page = undefined,
+		perPageArg = undefined,
```

Comment:
Looking at the previous `fetchLinkSuggestions` implementations, it looks like this should be `perPage` not `perPageArg`.

```suggestion
		perPage: perPageArg = undefined,
```

Reply:
Good catch, it was used to avoid to avoid redeclaring a const, but I shouldn't have added it to the function signiture. https://github.com/WordPress/gutenberg/pull/29993/files#diff-278a4a275170c4215a2f4e9d5c0c6d831f9328258220a76cda04b8d9f39f01a5L47